vue/sbc-common-components/src/store/account.ts

```diff
@@ -0,0 +1,122 @@
+      userSettings: [],
+      currentAccount: null,
+      currentAccountMembership: null,
+      pendingApprovalCount: 0,
+      currentUser: null
+    }
+  },
+  getters: {
+    accountName (): string {
+      return this.currentAccount && this.currentAccount.label
+    },
+    switchableAccounts (): UserSettings[] {
+      return this.userSettings?.filter(setting => setting.type === 'ACCOUNT')
+    },
+    username (): string {
+      return `${this.currentUser?.firstName || '-'} ${this.currentUser?.lastName || ''}`
+    }
+  },
+  actions: {
+    loadUserInfo () : KCUserProfile {
+      this.currentUser = KeyCloakService.getUserInfo()
+      return this.currentUser
+    },
+    async syncUserSettings (currentAccountId: string): Promise<UserSettings[]> {
+      const response = await AccountService.getUserSettings(this.currentUser?.keycloakGuid)
+      if (response?.data) {
+        const userSettings = response.data.filter(userSettings => (userSettings.type === 'ACCOUNT'))
+        const currentAccount = userSettings.find(org => String(org.id) === currentAccountId)
+        // if passed account is not user account list setting first one as current account
+        this.context.commit('setCurrentAccount', currentAccount || userSettings[0])
+        if (this.currentUser?.loginSource === LoginSource.BCSC ||
+          this.currentUser.roles.includes(Role.GOVMAccountUser)) {
+          await this.context.dispatch('fetchPendingApprovalCount')
+        }
+        this.userSettings = userSettings
+      } else {
+        this.userSettings = []
+      }
+      return this.userSettings
+    },
+    async fetchPendingApprovalAccount (): Promise<number> {
+      if (this.context.rootState.account?.currentAccount?.id) {
+        const response = await AccountService.getPendingMemberCount(
+          this.context.rootState.account.currentAccount.id, this.currentUser?.keycloakGuid)
+        this.pendingApprovalCount = (response && response.data && response.data.count) || 0
+      } else {
+        this.pendingApprovalCount = 0
+      }
+      return this.pendingApprovalCount
+    },
+    async syncCurrentAccount (userSetting: UserSettings): Promise<UserSettings> {
+      this.currentAccount = userSetting
+      ConfigHelper.addToSession(SessionStorageKeys.CurrentAccount, JSON.stringify(userSetting))
+      return this.currentAccount
+    },
+    async syncUserProfile () : Promise<KCUserProfile> {
+      // TODO improve the logic of not fetching the first name last name every time of header mounted
+      const response = await UserService.getUserProfile('@me')
+      if (response && response.data) {
+        const userProfile = response.data
+        // update the first name and last name for the users
+        const updateProfile:KCUserProfile = {
+          ...this.currentUser,
+          lastName: response.data.lastname,
+          firstName: userProfile.firstname
+        }
+        this.currentUser = updateProfile
+        return this.currentUser
```

Comment:
Seems weird to return something I set, but I wanted to be consistent with what was there before. I don't think the return for these values are used, but I'll have to check all of the projects.